core/autoscaling.go

```diff
@@ -536,6 +629,30 @@ func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) []str
 
 }
 
+func (a *autoScalingGroup) getPricetoBid(
+	baseOnDemandPrice float64, currentSpotPrice float64) float64 {
+
+	logger.Println("BiddingPolicy: ", a.region.conf.BiddingPolicy)
+
+	if a.region.conf.BiddingPolicy == "aggressive" {
+
+		if a.region.conf.SpotPriceBufferPercentage <= 0 {
+			a.region.conf.SpotPriceBufferPercentage = DefaultSpotPriceBufferPercentage
+		}
+
+		logger.Println("Launching spot instance with a bid =", math.Min(baseOnDemandPrice, currentSpotPrice*(1.0+a.region.conf.SpotPriceBufferPercentage/100.0)))
+		return math.Min(baseOnDemandPrice, currentSpotPrice*(1.0+a.region.conf.SpotPriceBufferPercentage/100.0))
+
+	} else if a.region.conf.BiddingPolicy == "normal" {
+
+		logger.Println("Launching spot instance with a bid =", baseOnDemandPrice)
+		return baseOnDemandPrice
+
+	}
+
+	return baseOnDemandPrice
+}
+
```

Comment:
I would also rewrite that as:
```go
func (a *autoScalingGroup) getPricetoBid(
	baseOnDemandPrice float64, currentSpotPrice float64) float64 {

	logger.Println("BiddingPolicy: ", a.region.conf.BiddingPolicy)

	if a.region.conf.BiddingPolicy == DefaultBiddingPolicy {
		logger.Println("Launching spot instance with a bid =", baseOnDemandPrice)
		return baseOnDemandPrice
	}

	logger.Println("Launching spot instance with a bid =", math.Min(baseOnDemandPrice, currentSpotPrice*(1.0+a.region.conf.SpotPriceBufferPercentage/100.0)))
	return math.Min(baseOnDemandPrice, currentSpotPrice*(1.0+a.region.conf.SpotPriceBufferPercentage/100.0))
}
```

Please note I removed the check of the value, as it seemed already done here: https://github.com/cristim/autospotting/pull/175/files#diff-de828bac595094756ea2a28fa415583bR219

My logic is always to avoid imbricated `if`, if there is an error check or direct return, then do it. I find that it usually simplifies reading/understanding of the code.